### PR TITLE
Nye tester implementert, metode implementert for å sjekke flere sider osv

### PR DESCRIPTION
**Pull request by bratne**
Løser issues [FKS-111], [FKS-113] , [FKS-197], and [FKS-191]

[FKS-111]: https://fintlabs.atlassian.net/browse/FKS-111?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[FKS-113]: https://fintlabs.atlassian.net/browse/FKS-113?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[FKS-197]: https://fintlabs.atlassian.net/browse/FKS-197?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[FKS-191]: https://fintlabs.atlassian.net/browse/FKS-191?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by olabrustad — APPROVED** [review]
Looks good.

File: src/main/java/no/fintlabs/AzureClient.java [hunk 1]
```diff
@@ -144,35 +142,93 @@ private void pullAllExtUsers() {
...
         );
         log.debug("*** <<< Done fetching all groups from AD ***");
     }
 
-    private void createGroup(AzureGroup azureGroup) {
-        log.debug("Azure create group: {}", azureGroup.getDisplayName());
-    }
+    //private void iterateGroupPages
+
+    public boolean doesGroupExist(String resourceGroupId) {
```
Comment (sourcecodeas) [inline src/main/java/no/fintlabs/AzureClient.java:161]:
```suggestion
    public boolean doesGroupExist(String resourceGroupId) {
    // Form the selection criteria for the MS Graph request
    String selectionCriteria = String.format("id,displayName,description,%s", configGroup.getFintkontrollidattribute());

    GroupCollectionPage groupCollectionPage = graphServiceClient.groups()
            .buildRequest()
            .select(selectionCriteria)
            .get();

    while (groupCollectionPage != null) {
        for (Group group : groupCollectionPage.getCurrentPage()) {
            JsonElement attributeValue = group.additionalDataManager().get(configGroup.getFintkontrollidattribute());
            
            if (attributeValue != null && attributeValue.getAsString().equals(resourceGroupId)) {
                return true; // Group with the specified ResourceID found
            }
        }

        // Move to the next page if available
        groupCollectionPage = groupCollectionPage.getNextPage() == null ? null :
            groupCollectionPage.getNextPage()
            .buildRequest()
            .select(selectionCriteria)
            .get();
    }

    return false; // Group with resourceID not found
}
```

Comment (sourcecodeas) [reply src/main/java/no/fintlabs/AzureClient.java:161]: Kan forenkles til dette

File: src/test/java/no/fintlabs/AzureClientTest.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        when(graphServiceClient.groups()).thenReturn(groupCollectionRequestBuilder);
+        when(graphServiceClient.groups().buildRequest()).thenReturn(groupCollectionRequest);
+
+        ResourceGroup resourceGroup = new ResourceGroup("12", "123", "testdisplayname", "testidpgroup", "testresourcename", "testresourcetype", "1000");
+        azureClient.addGroupToAzure(resourceGroup);
+
+        verify(groupCollectionRequest, times(1)).post(any(Group.class));
+    }
+
+    @Mock
```
Comment (sourcecodeas) [inline src/test/java/no/fintlabs/AzureClientTest.java:106]: flytt disse opp

**Review by sourcecodeas — APPROVED** [review]

**Review by sourcecodeas — APPROVED** [review]

**Review by olabrustad — APPROVED** [review]

**Pull request merged**